tenders/common/elf.h

```diff
@@ -25,10 +25,22 @@
 #ifndef COMMON_ELF_H
 #define COMMON_ELF_H
 
+/*
+ * guest_mprotect_fn() is called by the ELF loader to request that the page
+ * protection flags (prot) as used by the system mprotect(), i.e. PROT_X from
+ * sys/mman.h be applied to the guest memory range (addr_start .. addr_end).
+ * (t) is the backend-specific argument.
+ *
+ * Returns 0 on success, -1 and errno set on failure.
+ */
+typedef int (*guest_mprotect_fn_t)(void *t, uint64_t addr_start,
+        uint64_t addr_end, int prot);
+
```

Comment:
This lacks a description of `t` (see previous comment).

Reply:
It it correctly understood that in most cases `guest_mprotect_fn_t` primarily applies memory protection flags to the *host* view of the guest's memory mappings?

Reply:
> This lacks a description of t (see previous comment).

That's because I don't know what to call `t` in prose. "core tender struct" doesn't add any information.

> It it correctly understood that in most cases guest_mprotect_fn_t primarily applies memory protection flags to the host view of the guest's memory mappings?

Not quite. On KVM, at least the R/W part of the flags is applied also (indirectly) to the _guest_ mappings. On OpenBSD, if at some point if `vmm` gains support for `VMM_IOC_MPROTECT_EPT` then this function will also be responsible for directly manipulating the guest mappings via EPT.

Reply:
`t_arg` is fine with me, it's easier to search for and harder to confuse than `t`.